packages/cuba-react-ui/src/ui/form/EntityEditor.tsx

```diff
@@ -0,0 +1,244 @@
+import {FormattedMessage, injectIntl, WrappedComponentProps} from 'react-intl';
+import {
+  injectMainStore,
+  MainStoreInjected,
+  DataInstanceStore,
+  DataCollectionStore,
+  WithId,
+  isByteArray,
+  isOneToManyAssociation,
+} from "@cuba-platform/react-core";
+import {MetaPropertyInfo, MetaClassInfo, SerializedEntityProps} from "@cuba-platform/rest";
+import {Spinner} from "../Spinner";
+import {Field} from "./Field";
+import {MultilineText} from '../MultilineText';
+import {
+  clearFieldErrors,
+  constructFieldsWithErrors,
+  extractServerValidationErrors,
+  withLocalizedForm,
+} from '../..';
+import './EntityEditor.less';
+
+type EntityEditorProps = MainStoreInjected & WrappedComponentProps & {
+  entityName: string;
+  fields: string[];
+  dataInstance: DataInstanceStore<Partial<WithId & SerializedEntityProps>>;
+  associationOptions: Map<string, DataCollectionStore<Partial<WithId & SerializedEntityProps>>>;
+  onSubmit?: (fieldValues: {[field: string]: any}) => void;
+  onCancel: () => void;
+};
+
+@injectMainStore
+@observer
+class EntityEditorComponent extends React.Component<EntityEditorProps & FormComponentProps> {
+
+  @observable globalErrors: string[] = [];
+
+  reactionDisposers: IReactionDisposer[] = [];
+
+  componentDidMount(): void {
+    const {form, dataInstance, fields} = this.props;
+
+    this.reactionDisposers.push(reaction(
+      () => dataInstance.item,
+      () => {
+        form.setFieldsValue(
+          dataInstance.getFieldValues(fields)
+        );
+      },
+      {fireImmediately: true}
+    ));
+  }
+
+  componentWillUnmount(): void {
+    this.reactionDisposers.forEach(dispose => dispose());
+  }
+
+  get entityProperties(): MetaPropertyInfo[] {
+    const {entityName, fields, mainStore} = this.props;
+    return mainStore?.metadata ? getEntityProperties(entityName, fields, mainStore?.metadata) : [];
+  }
+
+  handleSubmit = (event: FormEvent) => {
+    const {form, intl, fields, onSubmit} = this.props;
+
+    event.preventDefault();
+    event.stopPropagation();
+
+    form.validateFields((clientError: any) => {
+      if (clientError) {
+        message.error(
+          intl.formatMessage({
+            id: "management.editor.validationError"
+          })
+        );
+        return;
+      }
+
+      if (onSubmit) {
+        onSubmit(form.getFieldsValue(fields));
+      } else {
+        this.defaultOnSubmit();
+      }
+    });
+  };
+
+  defaultOnSubmit = () => {
+    const {form, fields, dataInstance, intl} = this.props;
+
+    dataInstance
+      .update(form.getFieldsValue(fields))
+      .then(() => {
+        message.success(
+          intl.formatMessage({ id: "management.editor.success" })
+        );
+      })
+      .catch((serverError: any) => {
+        if (serverError.response && typeof serverError.response.json === "function") {
+          serverError.response.json().then((response: any) => {
+            clearFieldErrors(form);
```

Comment:
I prefer more compact code formatting for lines below, but it's on yours.

```suggestion
            clearFieldErrors(form);
            const { globalErrors, fieldErrors } = extractServerValidationErrors(response);
            this.globalErrors = globalErrors;
            
            if (fieldErrors.size > 0) {
              form.setFields(constructFieldsWithErrors(fieldErrors, form));
            }

            if (fieldErrors.size > 0 || globalErrors.length > 0) {
              message.error(intl.formatMessage({id: "management.editor.validationError"}));
            } else {
              message.error(intl.formatMessage({id: "management.editor.error"}));
            }
```